Add clippyx extension

0ssamaak0 opened the pull request:
## Description

<!-- A summary of your change. If you add a new extension or command, explain what it does. -->

## Screencast

<!-- If you add a new extension or command, include a screencast (or screenshot for straightforward changes). A good screencast will make the review much faster - especially if your extension requires registration in other services.  -->

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/clippyx/src/caption.tsx:
```diff
@@ -0,0 +1,136 @@
+  path: string;
+}
+
+interface Preferences {
+  threshold: number;
+  topK: number;
+}
```
You can remove this part, this is automaticlly generated in `raycast-env.d.ts`

0ssamaak0 replied:
I got this error when I tried removing it

```
src/caption.tsx:27
:30 - error TS2339: Property 'threshold' does not exist on type 'ExtensionPreferences'.

27       threshold: preferences.threshold,
                                ~~~~~~~~~

src/caption.tsx:28:26 - error TS2339: Property 'topK' does not exist on type 'ExtensionPreferences'.

28       top_k: preferences.topK,
```

pernielsentikaer reviewed (changes requested):
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Is there a reason why all command has the same preferences, shouldn't it be enough to set for the extension itself?

- I'm not sure what the preferences it for, does the end user knows for sure?

- Maybe update the README with a link to the download page and tell a bit about it

- Could you record me a screencast of the extensions flow?

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

0ssamaak0 commented:
> Hi 👋
> 
> Thanks for your contribution 💪
> 
> I have now tested your extension, and I have some feedback ready for you:
> 
> * Is there a reason why all command has the same preferences, shouldn't it be enough to set for the extension itself?
> * I'm not sure what the preferences it for, does the end user knows for sure?
> * Maybe update the README with a link to the download page and tell a bit about it
> * Could you record me a screencast of the extensions flow?
> 
> I'm looking forward to testing this extension again 🔥
> 
> Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.


- Yes, it's different since it uses different models / modalities
- The end user is already using CLIPPyX from the webui, but using it from raycast will be far more convinient
- I updated it and added link to the main repo
- here it is: 

https://github.com/user-attachments/assets/c1f1456a-0c4c-454a-ab36-3d5aafffcbaa

pernielsentikaer commented:
Should be better now 🙂

0ssamaak0 commented:
> Could you add at least one [screenshot](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) of the extension

Done ✅

Thank you for help

pernielsentikaer commented on extensions/clippyx/package.json:
```diff
@@ -0,0 +1,113 @@
+        {
+          "name": "topK",
+          "type": "textfield",
+          "required": true,
```
Would it make sense to set all `required` to `false`  instead so the user don't need to set them before using each command?

0ssamaak0 replied:
I think it's better to leave them `true`? they will be set in the first time only right?

pernielsentikaer reviewed (approved):
Hi 👋 

Looks good to me, approved 🔥
